packages/rich-text/src/plugins/CommandPalette/onKeyDown.ts

```diff
@@ -36,6 +36,10 @@ export const createOnKeyDown = (): KeyboardHandler => {
         }
       }
 
+      if (isHotkey('space', event)) {
```

Comment:
couldn't we close/remove it on every key except arrow keys and enter?

for example : `contentful.com/developers` would also have this issue

Reply:
Well we have a search enabled for rich text commands. Typing something after the slash will sort the entries based on what you typed

Reply:
So this would remove the search completely. Now it is only removed when you type a space

Reply:
How about escape only closes the command palette and keeps focus?

Reply:
Then the second time you press escape it closes the command palette?

Reply:
Not sure if I understood correctly, but this is exactly what they complained about, they don't want to use the escape button since it blurs the editor and they'd have to focus it again

Reply:
I mean change the behaviour so that we don't lose focus the first time escape is clicked, it might involved focusing again in code to the point they are currently at

Reply:
- /
- "some text"
- [ESCAPE] _closes command palette but keeps focus in editor_
- [ESCAPE] _loses focus_

Reply:
We would still have the issue where the user types content with a slash, does not want anything from the rich text commands palette, and keeps typing... the content is still split up:

<img width="268" alt="Screenshot 2023-01-31 at 16 04 50" src="https://user-images.githubusercontent.com/22968325/215797220-ac8af76e-c7d4-474b-995f-efe295aece11.png">

Reply:
https://github.com/contentful/field-editors/blob/7fbebc371800f12a12adbd7278c22449c6ff85f1/packages/rich-text/src/internal/misc.ts#L23 we have this function

Reply:
If we go that route I think we should have a clear UI that shows that the rich text command span is currently applied, and that they need to press ESC to remove it

Reply:
Maybe this works for using space inside a search:
```
if (isHotkey(['backspace', 'space'], event)) {
  // if it is the last character in the command string
  if (range.focus.offset - range.anchor.offset === 1) {
    removeMark(editor, COMMAND_PROMPT, range);
  }
}
 ```
 
-  `/` + `space` -> remove command prompt
- `/` + `entry editor` -> still open and can search

Reply:
This would only remove the mark when you type a space _right after_ the slash right?
So what if you type:

> You can use /slash to open the command palette.

This would still apply the two spans :/

Reply:
ah thats sad, it should probably also remove the mark if it is closed without applying content right?

Reply:
> We would still have the issue where the user types content with a slash, does not want anything from the rich text commands palette, and keeps typing... the content is still split up:

What's the issue with the split up? If it's bad we can normalize that in the editor or in the preprocessing stage before calling `onChange`. Also, Slate should automatically normalize that when the two text nodes share the same marks. See rule 2 [here](https://docs.slatejs.org/concepts/11-normalizing)